plugins/obs-ffmpeg/texture-amf.cpp

```diff
@@ -1294,9 +1295,16 @@ static bool amf_avc_init(void *data, obs_data_t *settings)
 	int64_t bf = obs_data_get_int(settings, "bf");
 
 	if (enc->bframes_supported) {
-		set_avc_property(enc, MAX_CONSECUTIVE_BPICTURES, 3);
+		set_avc_property(enc, MAX_CONSECUTIVE_BPICTURES, bf);
 		set_avc_property(enc, B_PIC_PATTERN, bf);
 
+		/* AdaptiveMiniGOP is suggested for some types of content such
+		 * as those with high motion.
+		 */
```

Comment:
Add a note about this only taking effect if Pre-Analysis is enabled.
```suggestion
		/* AdaptiveMiniGOP is suggested for some types of content such
		 * as those with high motion. This only takes effect if
		 * Pre-Analysis is enabled.
		 */
```

Reply:
Added